msrest/serialization.py

```diff
@@ -552,13 +546,14 @@ def body(self, data, data_type, **kwargs):
         # Just in case this is a dict
         internal_data_type = data_type.strip('[]{}')
         internal_data_type = self.dependencies.get(internal_data_type, None)
+        is_xml_model_serialization = kwargs.get("is_xml", False)
         if internal_data_type and not isinstance(internal_data_type, Enum):
             try:
                 deserializer = Deserializer(self.dependencies)
                 # Since it's on serialization, it's almost sure that format is not JSON REST
                 # We're not able to deal with additional properties for now.
                 deserializer.additional_properties_detection = False
-                if issubclass(internal_data_type, Model) and internal_data_type.is_xml_model():
+                if issubclass(internal_data_type, Model) and (internal_data_type.is_xml_model() or is_xml_model_serialization):
```

Comment:
Just check the is_xml_model_serialization, because is_xml_model is supposed to be checked before